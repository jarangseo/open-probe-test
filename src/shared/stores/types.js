export const SAVED_HISTORY = 'advanced/SAVED_HISTORY'