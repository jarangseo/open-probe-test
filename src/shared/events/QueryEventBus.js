import Vue from 'vue'

export const queryEventBus = new Vue({
  methods: {
    addTab (data) {
      this.$emit('addTab', data)
    }
  }
})