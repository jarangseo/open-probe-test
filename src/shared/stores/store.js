import Vue from 'vue'
import Vuex from 'vuex'
import { advancedModule } from './modules/advancedModule'

Vue.use(Vuex)

export default new Vuex.Store({
  modules: {
    advancedModule
  }
})
