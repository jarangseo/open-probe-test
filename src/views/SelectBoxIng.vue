<template>
  <div class="cus-select-box">
    <div class="select-head" v-on:click="toggle">
      <input type="text" class="selected-item" :value="selectedItemString" readonly>
    </div>
    <div class="select-body" v-show="open">
      <div class="search-wrap" style="display:none">
        <input type="text" class="search"/>
      </div>
      <div class="default-list" v-if="itemList.length > 0">
        <vue-scrollbar classes="list-area" ref="Scrollbar">
          <ul class="list-items">
            <li v-for="item in itemList" :key="item.index" @click="setSelectedItemIdx(item.index)">
              <a href="#">{{item.name}}</a>
            </li>
          </ul>
        </vue-scrollbar>
      </div>
    </div>
  </div>
</template>
<script>
import VueScrollbar from 'vue2-scrollbar'
import Icon from 'vue-awesome/components/Icon.vue'

export default{
  components: {
    VueScrollbar,
    Icon
  },
  props: {
    itemList: {
      type: Array,
      default: () => []
    }
  },
  data() {
    return {
      selectedItemIdx: 0,
      open: false
    }
  },
  computed: {
    selectedItemString() {
      if(this.itemList && this.itemList.length) {
        return this.itemList[this.selectedItemIdx].name
      }else{
        return ''
      }
    }
  },
  methods: {
    toggle() {
      this.open = !this.open
    },
    setSelectedItemIdx(idx) {
      console.log(idx)
      this.selectedItemIdx = idx
      this.toggle()
    }
  }
}
</script>

<style lang="scss">
  @import "~@/assets/styles/_variables.scss";
  //selectbox
  .cus-select-box {
    position: relative;
    display: inline-block;
    vertical-align: middle;
    &.cal-l {
      width:400px;
      .select-head {
        height: 36px;
        padding: 0 28px 4px 12px;
        &:after {
          width: 28px;
          height: 34px;
        }
        &:before {
          top:14px;
          right:11px;
          vertical-align: middle;
        }
      }
      .selected-item {
        height: 29px;
        font-size:14px;
      }
    }
    &.cal-m {
      width:300px;
      .selected-item {
        font-size:$default-form-text-size;
      }
    }
    &.cal-s {
      width:160px;
      .selected-item {
        font-size:$default-form-text-size;
      }
    }
    &.cal-param {
      width:140px;
      .selected-item {
        font-size:$default-form-text-size;
      }
    }
    &.open {
      .select-head {
        &:before {
          background: url("~@/assets/images/basic-img.png");
          background-position: -115px -12px;
        }
      }
    }
    .select-head {
      overflow: hidden;
      height: 28px;
      padding: 0 28px 4px 12px;
      background-color: $default-form-bgcolor;
      border-top:1px solid $form-bdtop-color;
      border-right:1px solid $form-bdright-color;
      border-bottom:1px solid $form-bdright-color;
      border-left:1px solid $form-bdleft-color;
      box-sizing: border-box;

      &.disabled {
        background-color:#e5e5e5 !important;
      }
      &:after {
        content: '';
        display: block;
        position: absolute;
        width: 22px;
        height: 26px;
        top: 0;
        right: 0;
        border-top:1px solid #d7d7d7;
        border-right:1px solid #c7c7c7;
        border-bottom:1px solid #c7c7c7;
        border-left:1px solid #d7d7d7;

      }
      &:before {
        content: '';
        display: inline-block;
        position: absolute;
        top:11px;
        right:8px;
        vertical-align: middle;
        width:9px;
        height:6px;
        background: url("~@/assets/images/basic-img.png");
        background-position: -115px -3px;
      }
    }
    .select-body {
      z-index:10;
      position: absolute;
      width:100%;
      margin-top: -1px;
      //border-width:1px 1px 1px 1px;
      border:1px solid #cacaca;
      background-color: #fafafa;
      box-sizing: border-box;
      .search-wrap {
        .search {
          width:100%;
          height: 27px;
          border-width:0 0 1px 0;
          border-style:solid;
          border-color:$selectbox-border-color;
          box-sizing: border-box;
        }
      }
    }

    .select-footer {
      height: 40px;
      padding-left: 12px;
      line-height: 40px;
      font-size:12px;
      border-top:1px solid #d9d9d9;
      > label {
        display: inline-block;
        width:inherit;
        padding: 0;
        text-decoration: underline;
        background-color: transparent;
        box-sizing: border-box;
      }
      .cb-counter {
        float: right;
        color: #95c8ff;
        margin-right: 8px;
        font-family:'Roboto';
        font-weight: normal;
        &:after{
          content: '';
          display: table;
          clear:both;
        }
      }
    }
    .list-area {
      width:100%;
      max-height: 182px;
      margin-bottom:6px;
      box-sizing: border-box;
      .list-items {
        width:100%;
        padding-top:6px;
        min-height: 40px;
        box-sizing: border-box;
        font-size: 12px;
        font-weight: bold;
        font-weight: 500;
      }
      .list-items > li {
        position: relative;
        a {
          display: inline-block;
          padding-right:20px;
          height: $default-form-height;
          line-height: $default-form-height;
          color:#86888c;
          box-sizing: border-box;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
          word-break: break-all;
          word-wrap: break-word;
        }
      }
    }

    // list -- width
    .default-list {
      .list-area {
        width:100%;
      }
    }
    .multiple-list {
      .list-area {
        width:100%;
      }
    }
    .description-list {
      .list-area {
        width:178px;
        display: inline-block;
        vertical-align: top;
      }
    }
    .description-view {
      .list-area {
        display: inline-block;
        width:100%;
        vertical-align: top;
        padding:6px 18px 0 14px;
        word-wrap:break-word;
        word-break: keep-all;
        box-sizing: border-box;
      }
    }

    .default-list .list-area .list-items > li a,
    .description-list .list-area .list-items > li a {
      padding-left: 12px;
      width:100%;
      &.on,
      &:hover{
        color:#32353b;
        background-color: $body-color;
      }
    }
    .multiple-list .list-area .list-items > li label {
      vertical-align: top;
      height: $default-form-height;
      line-height: $default-form-height;
      color:#86888c;
    }

    .description-view {
      display: inline-block;
      width:116px;
      height: 182px;
      border-left: 1px solid #cacaca;
      background-color: #fff;
      vertical-align: top;
      box-sizing: border-box;
      p {
        font-size:12px;
        color:#86888c;
        line-height: 22px;
        > span {
          font-weight: 500;
          display: block;
        }
      }
    }
    .no-result {
      width:100%;
      height: 170px;
      text-align: center;
      line-height: 170px;
    }
    .multiple-tag-area {
      overflow: hidden;
      position: absolute;
      width:257px;
      display: inline-block;
      height: 26px;
      white-space: nowrap;
    }
    .selected-tag {
      display: inline-block;
      margin:4px 4px 0 0;
      padding:3px 3px 2px 4px;
      font-size: 12px;
      font-weight: bold;
      color: #32353b;
      background-color: #dee3eb;
      > a {
        display: inline-block;
        margin:0 0 0 2px;
        padding:0;
        vertical-align: middle;
        color:#32353b;
        opacity: 0.4;
        &:hover {
          color:#32353b;
        }
        .ico-s {
          width:8px;
          height:8px;
        }
      }
    }
    .selected-item {
      height:22px;
      width:99%;
      margin:2px 0 -2px -1px;
      padding: 0 $default-form-padding 0 0;
      font-family: inherit;
      font-weight: 500;
      color: $basic-menu-active-color;
      border:none;
      background:transparent;
      &::placeholder{
        color:$dark-font-color;
      }
      &[disabled]{
        cursor: default;
        color: rgba(0,0,0,.3);
        background-color: $default-form-dimmed-bgcolor;
      }
    }
    .disabled {
      .selected-item {
        color: rgba(0,0,0,.3);
        &::placeholder {
          color: rgba(0,0,0,.3);
        }
      }
    }
    .ovrlay-body {
      z-index: 2;
      position: fixed;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
    }
  .vue-scrollbar__scrollbar-vertical {
    width: 8px;
    top: 0;
    right: 7px;
    bottom: 0;
    margin-bottom: 0px;
  }
  .nodata-text {
    margin: 9px 0 0 0;
    text-align: center;
    vertical-align: middle;
  }
  .nodata-text-nl {
    margin: 5px 0;
    text-align: center;
    vertical-align: middle;
    line-height: 18px;
  }
}// end cus-select-box
</style>
