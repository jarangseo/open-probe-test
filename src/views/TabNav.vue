<template>
  <div>
      <div class="advanced-tabmenu-wrap">
        <ul class="lnb-tabmenu-wrap">
          <li class="lnb-tab" @click="activeTab(0)">
            <a href="#" :class="{'selected':activeTabIdx === 0}">Table info</a>
          </li>
          <li class="lnb-tab" @click="activeTab(1)">
            <a href="#" :class="{'selected':activeTabIdx === 1}">My Temp</a>
          </li>
        </ul>
      </div>

      <div class="wrap_select">
        <select-box-ing :itemList="upperSelectorItems"></select-box-ing>
        <table-list></table-list>
      </div>
    </div>
</template>
<script>
import SelectBoxIng from './SelectBoxIng'
import TableList from './TableList'

export default {
  data() {
    return {
      activeTabIdx: 0,
      upperSelectorItems: [
        {
          name: 'Hive',
          index: 0,
          selected: true
        },
        {
          name: 'Impala',
          index: 1,
          selected: false
        }
      ],
    }
  },
  components: {
    SelectBoxIng,
    TableList
  },
  methods: {
    activeTab(idx) {
      this.activeTabIdx = idx
      console.log(idx)
    }
  }
}
</script>

<style lang="scss">
.advanced-tabmenu-wrap {
  display: table;
  table-layout: fixed;
  width: 100%;
  margin:0 0 10px 0;
  color: #adafb3;
  font-size:13px;
  overflow: hidden;
  .lnb-tab {
    float: left;
    display: table-cell;
    position: relative;
    width:50%;
    padding:0;
    text-align: center;
    font-family: 'Roboto';
    font-weight: 500;
    color: #adafb3;
    > a {
      display: block;
      height: 42px;
      line-height: 42px;
      background-color: #f2f4f7;
      box-sizing: border-box;
      color: #adafb3;
      &.selected {
        color: #32353b;
        background-color: #fff;
      }
    }
    &:after {
      content: ' ';
      display: table;
      clear:both;
    }
  }
}
.wrap_select {
  padding: 0 20px 20px 20px;
  border-bottom:1px solid #d1d8e4;
}
</style>