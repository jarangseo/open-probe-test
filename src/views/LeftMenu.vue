<template>
  <div>
    <div class="fixed-menu">
      <select-box-ing :itemList="upperSelectorItems"></select-box-ing>
      <ul class="menu-group">
        <li @click="viewSavedQueryTab(0)"><a href="#">Saved Query</a></li>
        <li @click="viewSavedQueryTab(1)"><a href="#">Recent Query</a></li>
      </ul>
    </div>
    <tab-nav></tab-nav>
  </div>
</template>
<script>
import * as types from '../shared/stores/types'
import SelectBoxIng from './SelectBoxIng'
import TableList from './TableList'
import Icon from 'vue-awesome/components/Icon.vue'
import TabNav from './TabNav'
import { queryEventBus } from '@/shared/events/QueryEventBus'

export default {
  components: {
    SelectBoxIng,
    TableList,
    TabNav,
    Icon
  },
  data() {
    return {
      upperSelectorItems: [
        {
          name: 'Hive',
          index: 0,
          selected: true
        },
        {
          name: 'Impala',
          index: 1,
          selected: false
        }
      ],
      queryMenuItems: [
        {
          name: 'Saved Query',
          index: 0,
          selected: false
        },
        {
          name: 'Recent Query',
          index: 1,
          selected: false
        }
      ],
      tabItems: [
        {
          name: 'Table info',
          index: 0,
          selected: true
        },
        {
          name: 'My Temp',
          index: 1,
          selected: false
        }
      ],
      searchInfo: {
        database: {
          selectedDBindex: 0,
          dbSearchString: ''
        },
        table:{
          tableSearchString: '',
          clickedTableIndex: -1
        }
      }
    }
  },
  methods: {
    viewSavedQueryTab(idx) {
      queryEventBus.addTab(idx)
    }
  },
  created() {
    
  },
  mounted() {
    
  }
}
</script>
<style lang="scss">
  .lnb {
    .cus-select-box {
      width:210px;
      max-width:210px;
      margin-bottom:8px;
      >.select-head {
        width:210px;
      }
    }
  }
  .fixed-menu {
    .cus-select-box {
      margin-left:20px;
      margin-bottom:10px;
    }
  }
  .tempdb {
    margin:0 auto 10px auto;
    width: 209px;
    color:#fff;
    background-color: #afb6bc;
    height: 28px;
    line-height: 28px;
    padding-left:31px;
    border:none;
    position: relative;
  }
  .wrap-refresh {
    position: relative;
    top: -36px;
    left: 10px;
    text-align: left;
    height: 0;
    line-height: 30px;
    > a {
      color:#fff;
    }
  }
  .loading-box {
    position:relative;
    width:100%;
    height:150px;
  }
  .nodata-text {
    margin:30px 0;
    text-align: center;
  }
  .table-list-area {
    margin-top:13px;
  }
  .table-name {
    font-weight: 500;
  }
  .no-data-area {
    position: relative;
    height: 180px;
    border: 1px solid #fff;
    .no-data-dsc {
      margin-top: 50%;
    }
  }
  .no-data-img-sm {
    width:45px;
    height:45px;
    background-position: -621px -54px !important;
    background: url("~@/assets/images/basic-img.png");
    margin:0 auto 40px auto;
  }
</style>
