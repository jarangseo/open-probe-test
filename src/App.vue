<template>
  <div class="container" id="app">
    <div class="wrap">
      <nav class="lnb">
        <left-menu></left-menu>
      </nav>
      <section class="content"></section>
    </div>
  </div>
</template>
<script>
import LeftMenu from './views/LeftMenu.vue'
import 'vue-awesome/icons'
import { queryEventBus } from '@/shared/events/QueryEventBus'

export default {
  components: {
    LeftMenu
  },
  methods: {

  },
  created() {
    queryEventBus.$on('addTab', (data) => {
      console.log(data)
    })
  }
}
</script>
<style lang="scss">
  @import "~@/assets/styles/settings.scss";
</style>
