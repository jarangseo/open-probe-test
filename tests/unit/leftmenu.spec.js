import { shallowMount, createLocalVue } from '@vue/test-utils'
import { mount } from '@vue/test-utils';
import LeftMenu from '@/views/LeftMenu.vue'
import SelectBoxIng from '@/views/SelectBoxIng'
import TabNav from '@/views/TabNav'
import { queryEventBus } from '@/shared/events/QueryEventBus'
import App from '@/App'
import Vuex from 'vuex'
import * as types from '@/shared/stores/types'

let LeftMenuWrapper;
let SelectBoxIngWrapper;
let TabNavWrapper;
let queryEventBusWrapper;
let AppWrapper;

let localVue = createLocalVue()
localVue.use(Vuex)
let actions
let store

beforeEach(() => {
  actions = {
    [types.SAVED_HISTORY]: jest.fn()
  }
  store = new Vuex.Store({
    state: {},
    actions
  })

  TabNavWrapper = shallowMount(TabNav, {
    localVue,
    store
  })
  expect(actions[types.SAVED_HISTORY]).toHaveBeenCalled();

  LeftMenuWrapper = shallowMount(LeftMenu)
  SelectBoxIngWrapper = shallowMount(SelectBoxIng)
  // queryEventBusWrapper = mount(queryEventBus)
  AppWrapper = shallowMount(App)
});

describe('LeftMenu.vue', () => {
  it('renders', () => {
    expect(LeftMenuWrapper.isVueInstance()).toBeTruthy()
    expect(SelectBoxIngWrapper.isVueInstance()).toBeTruthy()
  })
})

describe('[SelectBoxIng] Hive/Impala 변경', () => {
  it('UI는 동일', () => {
    const prevOpenBool = SelectBoxIngWrapper.vm.open
    SelectBoxIngWrapper.find('.select-head').trigger('click')
    expect(typeof SelectBoxIngWrapper.vm.toggle).toBe('function')
    expect(SelectBoxIngWrapper.vm.open).toBe(!prevOpenBool)
  })
  it('히스토리는 각각 저장됨', () => {
    const mockFn = jest.fn()
    SelectBoxIngWrapper.setMethods({
      toggle: mockFn
    })
    SelectBoxIngWrapper.find('.select-head').trigger('click')
    expect(mockFn).toBeCalled()
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})

describe('Menu영역의 Menu명 클릭시', () => {
  it('해당 Tab Menu가 없는 경우 > Contents영역이 Tab으로 추가', () => {
    const mockFn = jest.fn()
    queryEventBus.addTab = mockFn
    LeftMenuWrapper.find('.menu-group li').trigger('click')
    expect(mockFn).toBeCalled()
    expect(mockFn).toBeCalledWith(0)
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
  it('해당 Tab Menu가 있는 경우 > 활성화됨', () => {
    expect(SelectBoxIngWrapper).toMatchSnapshot()
    expect(LeftMenuWrapper).toMatchSnapshot()
  })
})

describe('Table Info > select', () => {
  it('select box 변경시 하단 Table List변경', () => {

  })
  it('검색어 입력하여 table list검색', () => {

  })
})

describe('Table Info > table list', () => {
  it('클릭 > Table내 컬럼을 fold/unfold', () => {

  })
  it('클릭 > 해당 Table의 Column 리스트 정보 없는경우 API 호출', () => {

  })
  it('더블 클릭 > Editor창 커서 위에 입력하되 형식은 Database.table형태', () => {

  })
  it('EYE 버튼 클릭 > 우 하단 Contants영역 Tab추가되며 Preview 노출', () => {
    const mockFn = jest.fn()
    queryEventBus.addTab = mockFn
    // LeftMenuWrapper.find('.menu-group li').trigger('click')
    expect(mockFn).toBeCalled()
    expect(mockFn).toBeCalledWith(0)
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
  it('Preview버튼 클릭 > 우 하단 Contants영역 Tab추가되며 Preview 노출', () => {
    const mockFn = jest.fn()
    queryEventBus.addTab = mockFn
    LeftMenuWrapper.find('.menu-group li').trigger('click')
    expect(mockFn).toBeCalled()
    expect(mockFn).toBeCalledWith(0)
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
  it('Column명 더블 더블클릭 > Column명을 Editor창 커서 위치에 입력', () => {

  })
})