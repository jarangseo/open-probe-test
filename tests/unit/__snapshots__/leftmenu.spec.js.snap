// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu영역의 Menu명 클릭시 해당 Tab Menu가 있는 경우 > 활성화됨 1`] = `
<div class="cus-select-box">
  <div class="select-head"><input type="text" readonly="readonly" class="selected-item"></div>
  <div class="select-body" style="display: none;">
    <div class="search-wrap" style="display: none;"><input type="text" class="search"></div>
    <!---->
  </div>
</div>
`;

exports[`Menu영역의 Menu명 클릭시 해당 Tab Menu가 있는 경우 > 활성화됨 2`] = `
<div>
  <div class="fixed-menu">
    <select-box-ing-stub itemlist="[object Object],[object Object]"></select-box-ing-stub>
    <ul class="menu-group">
      <li><a href="#">Saved Query</a></li>
      <li><a href="#">Recent Query</a></li>
    </ul>
  </div>
  <tab-nav-stub></tab-nav-stub>
</div>
`;
